docs/hub/how-to-upstream.md

```diff
@@ -0,0 +1,259 @@
+# How to integrate upstream methods in your library
+
+Now that you've seen more downstream methods for downloading files, it is time to introduce you to additional *upstream* methods. Upstream methods refers to functions that publish files to the Hub from your library. This guide will show you how to:
+
+* Use the `HfApi` class to manage a repository.
+* Use the `Repository` class to manage a repository with Git-like commands.
+
+## `HfApi`
+
+The `HfApi` class is a low level class that wraps around HTTP requests. There are many useful tasks you can accomplish with the `HfApi` class including: 
+
+- List and filter models.
+- Inspect model or dataset metadata.
+- Delete a repository.
+- Change the visibility of a repository.
+
+### List and filter
+
+It can be helpful for users to see a list of available models, and filter those models according to a specific language or framework. Use the `list_models` function with the `filter` parameter to search for a model of interest:
```

Comment:
Maybe this is a good place to mention that if you're an owner of a library or organization, you can use this to get all your library models. Maybe even throw a link to https://huggingface.co/docs/hub/endpoints#webhook in here

Reply:
Good call! This will explicitly show how library and organization owners can get value out of `list_models`. I'm not too sure how the webhook is related to listing and filtering models though. 😅 Is it just so owners can be alerted when a model is added to their library/org?